pkg/admin/facade/oauth.go

```diff
@@ -0,0 +1,98 @@
+package facade
+
+import (
+	"github.com/authgear/authgear-server/pkg/api/apierrors"
+	"github.com/authgear/authgear-server/pkg/lib/authn/authenticationinfo"
+	"github.com/authgear/authgear-server/pkg/lib/config"
+	"github.com/authgear/authgear-server/pkg/lib/oauth"
+	"github.com/authgear/authgear-server/pkg/lib/oauth/handler"
+	"github.com/authgear/authgear-server/pkg/lib/oauth/protocol"
+	"github.com/authgear/authgear-server/pkg/util/clock"
+)
+
+type OAuthAuthorizationService interface {
+	CheckAndGrant(
+		clientID string,
+		userID string,
+		scopes []string,
+	) (*oauth.Authorization, error)
+}
+
+type OAuthTokenService interface {
+	IssueOfflineGrant(
+		client *config.OAuthClientConfig,
+		opts handler.IssueOfflineGrantOptions,
+		resp protocol.TokenResponse,
+	) (*oauth.OfflineGrant, error)
+	IssueAccessGrant(
+		client *config.OAuthClientConfig,
+		scopes []string,
+		authzID string,
+		userID string,
+		sessionID string,
+		sessionKind oauth.GrantSessionKind,
+		resp protocol.TokenResponse,
+	) error
+}
+
+type OAuthFacade struct {
+	Config         *config.OAuthConfig
+	Authorizations OAuthAuthorizationService
+	Tokens         OAuthTokenService
+	Clock          clock.Clock
+}
+
+func (f *OAuthFacade) CreateSession(clientID string, userID string) (protocol.TokenResponse, error) {
```

Comment:
Better check if the user exists before generating the token, when I input an invalid user id I got the following error.

`pq: insert or update on table \"_auth_oauth_authorization\" violates foreign key constraint \"_auth_oauth_authorization_user_id_fkey\"`